p2p/src/sync/peer_v1.rs

```diff
@@ -616,48 +612,44 @@ where
         }
 
         // The first header must be connected to a known block (it can be in
-        // the chainstate, pending_headers or requested_blocks).
+        // the chainstate or requested_blocks).
         let first_header_prev_id = *headers
             .first()
             // This is OK because of the `headers.is_empty()` check above.
             .expect("Headers shouldn't be empty")
             .prev_block_id();
 
+        // Note: we require a peer to send headers starting from a block that we already have
+        // in our chainstate or from one that we've already requested from the peer.
```

Comment:
This "or from one that we've already requested from the peer" concerns me. Chainstate is the only source of truth.